daemon/media_socket.c

```diff
@@ -911,39 +911,41 @@ int __get_consecutive_ports(GQueue *out, unsigned int num_ports, unsigned int wa
 }
 
 /* puts a list of "struct intf_list" into "out", containing socket_t list */
-int get_consecutive_ports(GQueue *out, unsigned int num_ports, const struct logical_intf *log,
-		const str *label)
+int get_consecutive_ports(GQueue *out, unsigned int num_ports, struct call_media *media)
 {
 	GList *l;
 	struct intf_list *il;
 	const struct local_intf *loc;
+	const struct logical_intf *log = media->logical_intf;
+	const sockfamily_t *desired_family = media->desired_family;
+	const str *label = &media->call->callid;
 
 	for (l = log->list.head; l; l = l->next) {
 		loc = l->data;
 
+		// check desired family of local interface
+		if (strcmp(desired_family->rfc_name, loc->spec->local_address.addr.family->rfc_name) != 0)  {
```

Comment:
You can directly compare the "family" pointers with == or != since these are static structs. No need to use strcmp.

This is off the top of my head, but the `log` you get here should depend on what `desired_family` was, and the first interface in the `log` list should always be of the correct family. So I _think_ this check can be skipped altogether.

Reply:
Thanks for the feedback!

I've been looking into this and I have trouble deciding for port allocation based on ICE presence, for `other_media`, since ICE info comes later, in "answer", whereas ports get already allocated at "offer" time, in `__wildcard_endpoint_map`. I also tried forcing `__get_endpoint_map` for `other_media`, at "answer" time, and (all) ports get allocated, but ICE info is not updated in the SDP of 200OK.

I wonder if this simpler approach is acceptable:
1. Add a new config parameter "save_local_ports", default disabled
2. If disabled, nothing changes in port allocation logic
3. If enabled, alloc ports only on first available local interface, of desired_family
4. Document new config parameter

Thus existing ICE behavior should not change by default.

Thank you,
Stefan

Reply:
> Thanks for the feedback!
> 
> I've been looking into this and I have trouble deciding for port allocation based on ICE presence, for `other_media`, since ICE info comes later, in "answer", whereas ports get already allocated at "offer" time, in `__wildcard_endpoint_map`. I also tried forcing `__get_endpoint_map` for `other_media`, at "answer" time, and (all) ports get allocated, but ICE info is not updated in the SDP of 200OK.

Hm, is that so? AFAICT the `ICE` flag for both sides of the call should be available and properly set after `__ice_offer` - For the `other_media` it simply depends on what attributes were set in the SDP, and for `media` it depends on the SDP and what option flags were given. But I haven't looked at the whole thing in detail yet so I could be wrong of course.

> 
> I wonder if this simpler approach is acceptable:
> 
> 1. Add a new config parameter "save_local_ports", default disabled
> 2. If disabled, nothing changes in port allocation logic
> 3. If enabled, alloc ports only on first available local interface, of desired_family
> 4. Document new config parameter
> 
> Thus existing ICE behavior should not change by default.

Ok as a workaround - I'd prefer a more versatile and complete solution, but we can use this as a stepping stone, and then maybe I can finish it off when I have the time to play with it.